packages/scripts/config/webpack.config.js

```diff
@@ -177,12 +173,41 @@ const config = {
 					},
 				],
 			},
+			{
+				test: /\.svg$/,
+				use: [ '@svgr/webpack', 'url-loader' ],
+			},
+			{
+				test: /\.(bmp|png|jpe?g|gif)$/i,
+				// "url" loader works like "file" loader except that it embeds assets
+				// smaller than specified limit in bytes as data URLs to avoid requests.
+				loader: require.resolve( 'url-loader' ),
+				options: {
+					limit: 5000,
```

Comment:
We can either remove this limit and use files only, but we can also tweak this value.

Reply:
My experience when testing this was that no matter the limit, as long as the image was base64 encoded instead of being a file, the size increase of the css was always larger than the actual image file size.

I also think the browser needs to work more with painting the base64 encoded image then an image file. But that is mostly me guessing 😅

Thanks for looking into this. I never got the time to come up with this pr

Reply:
Gutenberg doesn't process static assets this way so I have no idea. That's why I'm waiting for feedback from folks that have some experience 😄

Reply:
@kimmenbert Due to how base64 encoding works the resultant inline image will be bigger than the original image file size, but the trade off is reducing the number total requests.

Keeping the limit to small (5kb) images or less is probably reasonable trade-off for increased CSS to outweigh the extra network call to an individual file.

Reply:
@aristath, you are working on similar explorations for CSS in core. What would you recommend here?

Reply:
> @aristath, you are working on similar explorations for CSS in core. What would you recommend here?

I'm still catching up so I don't know if this has already been implemented somewhere, but ideally, we'd have path resolution in the `theme.json` file. 
Then we'd be able to define the files as CSS-variables in there, and use the vars in our CSS.
In an HTTP/2 and beyond era it would be more efficient to load multiple smaller files than bundle images inside the main stylesheet.
base64-encoding is possible, but the result is always inefficient and resource-intensive both for network traffic and rendering since the end result is slightly larger in byte-size and the browser needs to decode it before rendering.
Inlining images in CSS would be something I'd only recommend for small SVG images. They don't need to be encoded and they are usually < 1kb (actually most times they're no more than a few bytes if we're talking about small background vectors).

Reply:
This config isn't targeting Gutenberg itself. It's for plugins and themes that want to use the same tools to build their products. I don't know if what you shared is relevant in this context but as I understand we should rather avoid inlining images into CSS. I will update code accordingly. We can always change that later, it's less important part of this proposal.

Reply:
efaa301 changes the loader to `file-loader` so it works the same way as for fonts. No more inlining for start.